typing/printtyp.ml

```diff
@@ -32,40 +32,257 @@ let rec longident ppf = function
   | Ldot(p, s) -> fprintf ppf "%a.%s" longident p s
   | Lapply(p1, p2) -> fprintf ppf "%a(%a)" longident p1 longident p2
 
-(* Print an identifier *)
+(* Print an identifier avoiding name collisions *)
 
-let unique_names = ref Ident.empty
+(* printing environment for path shortening and naming *)
+let printing_env = ref Env.empty
+let human_unique n id = Printf.sprintf "%s/%d" (Ident.name id) n
+
+module Namespace = struct
+
+  type t = [`Type|`Module|`Module_type|`Class|`Class_type|`Other]
```

Comment:
Why use a polymorphic variant here?

Reply:
I think the main reason was because `env.ml` is already using polymorphic variant for namespaces (when handling shadowing). I can revert to simple variants.